syntax = "proto3";

service Comunicator {

  rpc AddWorker (Message) returns (Message) {}
  rpc RemoveWorker (Number) returns (Message) {}
  rpc ListWorker (Message) returns (Message) {}
  rpc SubmitTask (Job) returns (Message) {}
}

message Number {
  uint32 amount = 1;
}

message Job {
  string programName = 1;
  string url = 2;
}

message Message {
  string message = 1;
}